events/attributevalue.go

```diff
@@ -18,6 +18,73 @@ type DynamoDBAttributeValue struct {
 	dataType DynamoDBDataType
 }
 
+// AttributeValue is the data for an attribute that has a name-value pair. The name is the data type, and the value is the actual data.
+// This is being replicated from "https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/#AttributeValue"(AWS SDK) to prevent dependency bloating or potential dependency import cycles.
```

Comment:
https://github.com/aws/aws-lambda-go/blob/d56e45b306530c767d46373096c2000d3478284f/events/attributevalue.go#L21-L23

How does replicating the type enable it's use with (dynamodbattribute.UnmarshalMap)[https://github.com/aws/aws-sdk-go/blob/f26268b77ea9619d95b1d04b6f3e3c5a03b61110/service/dynamodb/dynamodbattribute/decode.go#L86]? A test case would be helpful in understanding the usage.

Reply:
Seconded: I'd really like to be able to use `UnmarshalMap` in my lambda handlers

Reply:
What if we wrote, in perhaps another package to reduce the bloat of this package, a translator that translated DynamoDBAttributeValue to a dynamodb.*AttributeValue ? Then, if you needed such functionality, you could import that package and use it.

Reply:
In an  aid to those googling, there's some discussion of this here: https://stackoverflow.com/questions/49129534/unmarshal-mapstringdynamodbattributevalue-into-a-struct/50164289#50164289?newreg=c2e95160086c4a21a406a2e6ed0d15d9